File: common/src/main/java/com/pluxity/asset/entity/AssetCategory.java
package com.pluxity.asset.entity;

import com.pluxity.category.entity.Category;
import jakarta.persistence.*;
import java.util.ArrayList;
import java.util.List;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "asset_category")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class AssetCategory extends Category<AssetCategory> {

    @OneToMany(mappedBy = "category")
    private final List<Asset> assets = new ArrayList<>();

    @Column(name = "code", unique = true, length = 50)
    private String code;

    @Column(name = "icon_file_id")
    private Long iconFileId;

    @Builder
    public AssetCategory(String name, String code, Long iconFileId) {
        this.name = name;
        this.code = code;
        this.iconFileId = iconFileId;
    }

    public void updateIconFileId(Long iconFileId) {
        this.iconFileId = iconFileId;
    }

    public void updateCode(String code) {
        this.code = code;
    }

    public void addAsset(Asset asset) {
        if (asset != null && !this.assets.contains(asset)) {
            this.assets.add(asset);
        }
    }

    public void removeAsset(Asset asset) {
        if (asset != null) {
            this.assets.remove(asset);
        }
    }

    @Override
    public int getMaxDepth() {
        return 1;
    }
}

File: common/src/main/kotlin/com/pluxity/asset/entity/AssetCategory.kt
package com.pluxity.asset.entity

import com.pluxity.category.entity.Category
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.OneToMany
import jakarta.persistence.Table

@Entity
@Table(name = "asset_category")
class AssetCategory(
    @Column(name = "code", unique = true, length = 50, nullable = false)
    var code: String,
    @Column(name = "icon_file_id")
    var iconFileId: Long? = null,
    categoryName: String = "",
) : Category<AssetCategory>(categoryName) {
    @OneToMany(mappedBy = "category")
    val assets: MutableList<Asset> = mutableListOf()

    fun updateIconFileId(iconFileId: Long?) {
        this.iconFileId = iconFileId
    }

    fun updateCode(code: String) {
        this.code = code
    }

    fun addAsset(asset: Asset) {
        if (!this.assets.contains(asset)) {
            this.assets.add(asset)
        }
    }

    fun removeAsset(asset: Asset) {
        this.assets.remove(asset)
    }

    override val maxDepth: Int
        get() = MAX_DEPTH

    companion object {
        const val MAX_DEPTH = 1
    }
}
